### PR TITLE
Document open ports (required, monitoring, restricted)

### LINKED ISSUE #188: Document required open ports

### PR DESCRIPTION
**Pull request by SpComb**
Fixes  #188

Document all open ports on master/worker nodes: authenticated/required open ports, monitoring ports that are either restricted or should be restricted, and restricted ports that must not be exposed.

Most of the monitoring ports have already been restricted, but some of those are still open...

[Preview](https://github.com/kontena/pharos-cluster/blob/docs/open-ports/README.md#required-open-ports)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -42,6 +42,48 @@ Pharos Cluster executable can be downloaded from [https://github.com/kontena/pha
 - Minimal Ubuntu 16.04 (amd64 / arm64) hosts with SSH access
 - A user with passwordless sudo permission (`echo "$USER ALL=(ALL) NOPASSWD:ALL" | sudo tee /etc/sudoers.d/$USER`)
 
+### Required Open Ports
```
[the remark this answers is not in the document]
Comment (jakolehm) [reply README.md]: I think this is fine. We need to move docs into separate repo soon (README starts to be too long already).

File: README.md [hunk 2]
```diff
@@ -42,6 +42,48 @@ Pharos Cluster executable can be downloaded from [https://github.com/kontena/pha
...
+
+| Protocol    | Port        | Service         | Hosts / Addon         | Notes
+|-------------|-------------|-----------------|-----------------------|-------
+| TCP         | 22          | SSH             | All                   | authenticated management channel for `pharos-cluster` operations using SSH keys
+| TCP         | 6443        | kube-apiserver  | Master                | authenticated kube API for `pharos-cluster`, `kubectl` and worker node `kubelet` access using kube API tokens, RBAC
+| TCP         | 6783        | weave control   | All (weave)           | authenticated Weave peer control connections using the shared weave secret
+| UDP         | 6783        | weave dataplane | All (weave)           | authenticated Weave `sleeve` fallback using the shared weave secret
+| UDP         | 6784        | weave dataplane | All (weave)           | unauthenticated Weave `fastdp` (VXLAN), only used for peers on `network.trusted_subnets` networks
+| ESP (IPSec) |             | weave dataplane | All (weave)           | authenticated Weave `fastdp` (IPsec encapsulated UDP port 6784 VXLAN) using IPSec SAs established over the control channel
```
Comment (SpComb) [inline README.md]: Dunno if this is actually true, or how exactly the IPSec SAs are chosen

**Comment by SpComb** [comment]
Still going through some of the open monitoring ports to figure out what they expose.

**Comment by SpComb** [comment]
Also went through the various unauthenticated monitoring ports to assure myself that they're either locked down or sufficiently harmless to leave open by default... although they should ideally still be locked down even further.

Should be good to merge.

**Pull request merged**